dataStructures/queues/queue.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Queue data structure, it will be a list of Orders
+type Queue []*Order
+
+// Order data type
+type Order struct {
+	priority     int
+	quantity     int
+	product      string
+	customerName string
+}
+
+// New method creates a new order with required properties
+func (order *Order) New(priority, quantity int, product, customerName string) {
```

Comment:
Although Go doesn't have constructors in the same way as objective-oriented code does, we still call these `NewT()` functions constructors. Here's some things to consider:

- Include the name of the type, so this should be called `NewOrder()`
- It is OK to call the constructor `New()` only when the package name defines the type. For example, if this file was `package queue` rather than `package main` and the constructor was for a `Queue` then it is OK to call it `New()` because users of your package will call `queue.New()` and it reads really nicely but `queue.NewQueue()` repeats itself.
- Constructors should return new values rather than fill an existing one; right now you have to do this:

```
o := &Order{}
o.New(1, 2, "product", "customer)
```

It's much easier to use if you change to this:
```
func NewOrder(priority, quantity int, product customerName string) *Order {
  return &Order{
    priority: priority,
    …
  }
}

o := NewOrder(1, 2, "product", "customer") // All in one line :)
```

Reply:
@aschlosberg Wow! Thanks for this catch, I understand.